source/server/http/admin.cc

```diff
@@ -529,15 +545,77 @@ Http::Code AdminImpl::handlerClusters(absl::string_view url, Http::HeaderMap& re
   return Http::Code::OK;
 }
 
-// TODO(jsedgwick) Use query params to list available dumps, selectively dump, etc
-Http::Code AdminImpl::handlerConfigDump(absl::string_view, Http::HeaderMap& response_headers,
-                                        Buffer::Instance& response, AdminStream&) const {
-  envoy::admin::v2alpha::ConfigDump dump;
+void AdminImpl::addAllConfigToDump(envoy::admin::v2alpha::ConfigDump& dump,
+                                   const absl::optional<std::string>& mask) const {
+  for (const auto& key_callback_pair : config_tracker_.getCallbacksMap()) {
+    ProtobufTypes::MessagePtr message = key_callback_pair.second();
+    RELEASE_ASSERT(message, "");
```

Comment:
Why is this a release assert? If it's an invariant, it should just be regular ASSERT. Otherwise we need proper error handling. It should never be valid to crash the server due to a bad admin request.

Reply:
I copied it from the original [code](url) https://github.com/envoyproxy/envoy/blob/master/source/server/http/admin.cc#L540

It is not obvious to me why the pointer needs to be released, as the callbacks return new unique_ptrs ([example](https://github.com/envoyproxy/envoy/blob/master/source/common/upstream/cluster_manager_impl.cc#L867)). So, I've removed the release assert. That being said, I am not super confident in that choice because my familiarity with smart pointers is very low.

Reply:
The release here is a bit of a misnomer; this isn't about managing memory ownership but instead about release vs. debug builds. `RELEASE_ASSERT` applies in both, whereas `ASSERT` is only for debug builds.

Reply:
D'oh, thanks!